plugins/template/src/lib/TemplateCode.cpp

```diff
@@ -0,0 +1,26 @@
+#include "TemplateCode.h"
+#include "Filenames.h"
+#include <utility>
+
+namespace Template
+{
+    using VmiCore::ActiveProcessInformation;
+    using VmiCore::Plugin::PluginInterface;
+
+    TemplateCode::TemplateCode(VmiCore::Plugin::PluginInterface* pluginInterface,
+                               std::shared_ptr<VmiCore::IIntrospectionAPI> lowLevelIntrospectionApi)
+        : pluginInterface(pluginInterface),
+          logger(this->pluginInterface->newNamedLogger(TEMPLATE_LOGGER_NAME)),
+          lowLevelIntrospectionApi(std::move(lowLevelIntrospectionApi))
+    {
+        logger->bind({{VmiCore::WRITE_TO_FILE_TAG, LOG_FILENAME}}); // standard logger initialization
+    }
+
+    void TemplateCode::doStuffWithProcessStart(const ActiveProcessInformation& processInformation)
+    {
+        auto dtbValue = lowLevelIntrospectionApi->read64PA(processInformation.processCR3);
+        // example for structured logging calls
+        logger->info("Value @ Dtb {#x} (processId{}): {#x}",
```

Comment:
There is nothing here to interpret the curly braces. Looks to me like you are thinking of C# here ;)